packages/espressocash_app/lib/features/outgoing_link_payments/models/outgoing_link_payment.dart

```diff
@@ -90,4 +91,30 @@ class OLPStatus with _$OLPStatus {
     required BigInt slot,
     required EscrowPrivateKey escrow,
   }) = OLPStatusCancelTxSent;
+
+  /// Tx was recovered from the blockchain.
+  /// These transactions cannot be resent, they can only be cancelled
+  const factory OLPStatus.recovered({
+    required EscrowPublicKey escrowPubKey,
+  }) = OLPStatusRecovered;
+
+  /// Similar to [OLPStatus.cancelTxCreated], but only contains public key
+  const factory OLPStatus.recoveredCancelTxCreated(
```

Comment:
Canceling a recovered payment shouldn't differ from canceling a normal payment. What about moving `EscrowPrivateKey` out of the `OLPStatus` into `OutgoingLinkPayment` and making it nulable? In that case, recovered payments will just differ in having a private key null.